### ISSUE TITLE
How should extra-index-url be dealt with?

### ISSUE THREAD

**Issue by arruda** [kennethreitz/pipenv#221]
At first I added this as a comment in #34, but since it was an closed issue I don't think that was the right approach =/
I was thinking how can pipenv deal with extra-index-url, that is, one can't just assume this "source" will have all pypi sources and some more. For now I don't think Pipfile treats extra-index-url in a more clear way, so one way of dealing with this, would be considering that any source specified in the project sources (pipenv would have to be able to use all of the defined sources) could be a extra-index-url.
 
I imagined that one could use a Pipfile like:

```
[[source]]
url = "https://pypi.python.org/simple"
verify_ssl = true

[[source]]
url = "https://my.private.pypi/simple"
verify_ssl = true
```

Then whenever pipenv would access/use `project.source` as a single source, it would use instead a `project.sources` and do a re-try(if the given package can't be found in the first check for example) for each source present in `project.sources` and raise an error only if the given package couldn't be installed/searched/etc after trying every source available.

This would end the problem with `--extra-index-url` and would also make pipenv work with more then one source, instead forwarding the  `-i` when calling `pip`.

What do you think?

**Comment by arruda** [comment]
Just got a clarification that extra-index-url will be described as normal `[[sources]]` in Pipfile: https://github.com/pypa/pipfile/issues/68

(also I accidentally closed this issue, so I'm reopening it)

**Comment by nateprewitt** [comment]
Resolved with #222.
